language/move-prover/bytecode/src/compositional_analysis.rs

```diff
@@ -58,19 +58,21 @@ impl<'a> SummaryCache<'a> {
 
 /// Trait that lifts an intraprocedural analysis into a bottom-up, compositional interprocedural
 /// analysis. The derived `summarize` function computes a postcondition for the procedure under
-/// analysis, then stores it in the summary cache for other procedures to use.
-// TODO: allow client analyses to customize this by providing a function that converts a
-// postcondition into a summary
+/// analysis, then stores it in the summary cache for other procedures to use. The function
+/// 'summarize_map' is the same as 'summarize' but with the additional argument 'map' that is
+/// applied to the postcondition; the result of this function application is stored in the summary
+/// cache in place of the computed postcondition.
 pub trait CompositionalAnalysis: DataflowAnalysis
 where
     Self::State: 'static,
 {
-    fn summarize(
+    fn summarize_map<T:std::any::Any>(
         &self,
         func_env: &FunctionEnv<'_>,
         initial_state: Self::State,
         mut data: FunctionData,
-    ) -> FunctionData {
+        map : fn(Self::State) -> T
```

Comment:
Nit: `to_summary` or similar might be a more descriptive name.

Reply:
I changed to abstrct for the time since we are now requiring T to have trait AbstractDomain and so this is now an abstraction function. Happy to change to `to_summary` if you think that is better/more intuitive.

Reply:
I don't love the name "abstract" because it connotes precision loss, but this operation may or may not lose precision. For example: one common use-case for this might be performing an expensive (but precision-preserving) normalization step on the abstract state to keep the summary compact.